src/main/java/org/threadly/litesockets/UDPClient.java

```diff
@@ -180,6 +176,31 @@ public boolean setSocketOption(final SocketOption so, final int value) {
   public ClientOptions clientOptions() {
     return uso;
   }
+  
+  @Override
+  protected void addReadBuffer(final ByteBuffer bb) {
+    addReadStats(bb.remaining());
+    synchronized(readerLock) {
+      readBuffers.add(bb);
+      callReader();
```

Comment:
Do we want to call the readers while holding this lock?  If so, why?  Might be worth a comment in any case.